core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2401,8 +2390,13 @@ class ReplicaManager(val config: KafkaConfig,
         val lazyOffsetCheckpoints = new LazyOffsetCheckpoints(this.highWatermarkCheckpoints.asJava)
         val leaderChangedPartitions = new mutable.HashSet[Partition]
         val followerChangedPartitions = new mutable.HashSet[Partition]
+        val removeMirrorTopics = new mutable.HashSet[String]
         if (!localChanges.leaders.isEmpty) {
-          applyLocalLeadersDelta(leaderChangedPartitions, delta, lazyOffsetCheckpoints, localChanges.leaders.asScala, localChanges.directoryIds.asScala)
+          applyLocalLeadersDelta(leaderChangedPartitions, delta, lazyOffsetCheckpoints, localChanges.leaders.asScala, localChanges.directoryIds.asScala, removeMirrorTopics)
+          if (removeMirrorTopics.nonEmpty) {
+            info("!!! sent bumpLeaderEpoch:" + removeMirrorTopics)
+            mirrorMetadataManager.get.maybeUpdateLeaderEpoch(removeMirrorTopics.toList.asJava)
```

Comment:
I think we have a race here. This is called asynchronously after makeLeader. If the broker crashes between these calls, epoch might not bump.

Reply:
Yes, you're right! That's to be improved in the following PR using the state machine.